sklearn/preprocessing/_discretization.py

```diff
@@ -126,7 +127,7 @@ class KBinsDiscretizer(TransformerMixin, BaseEstimator):
     """
 
     @_deprecate_positional_args
-    def __init__(self, n_bins=5, *, encode='onehot', strategy='quantile',
+    def __init__(self, n_bins='auto', *, encode='onehot', strategy='quantile',
```

Comment:
should we call it `'sturges'` instead of `'auto'`?

Reply:
When I was experimenting I called it auto, since I used three different rules: Sqrt, Rice, Sturges. 
Same logic applies here: maybe in the future the formula will be change and we will have to change 'sturge' option to something different which could also lead to changing docs, tests etc. Calling it 'auto' and just changing the description is more sustainable imo